vpr/src/route/connection_router.cpp

```diff
@@ -649,27 +755,47 @@ void ConnectionRouter<Heap>::evaluate_timing_driven_node_costs(t_heap* to,
 
     float total_cost = 0.;
     const t_conn_delay_budget* delay_budget = cost_params.delay_budget;
-    if (delay_budget) {
+
+    if (run_rcv && to->path_data != nullptr) {
+        to->path_data->backward_delay += cost_params.criticality * Tdel;
+        to->path_data->backward_cong += (1. - cost_params.criticality) * get_rr_cong_cost(to_node, cost_params.pres_fac);
+
+        float expected_delay; // = router_lookahead_.get_expected_delay(to_node, target_node, cost_params, to->R_upstream);
+        float expected_cong;  // = router_lookahead_.get_expected_cong(to_node, target_node, cost_params);
+
+        std::tie(expected_delay, expected_cong) = router_lookahead_.get_expected_delay_and_cong(to_node, target_node, cost_params, to->R_upstream);
+
+        float expected_total_delay_cost; // = new_costs.backward_cost + cost_params.astar_fac * expected_cost;
+        float expected_total_cong_cost;
+
+        float expected_total_cong = cost_params.astar_fac * expected_cong + to->path_data->backward_cong;
```

Comment:
Should we consider finding a SIMD match for these?

Reply:
VFMADDxxxPS for example?

Reply:
I'm not attesting to have a lot of experience with this, but I thought SIMD was only profitable on large vectorized data sets?